### PR TITLE
Bug fix/ginger ops

### PR DESCRIPTION
**Pull request by AmanPrasad43**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Release Notes

- **New Features**
  - Enhanced title generation for the applications grid for better organization.
  - Introduced a mechanism to track item deletion status in the grid.

- **Improvements**
  - Simplified logic for setting the grid title.
  - Improved deletion logic to ensure better management of items in the grid.
  - Adjusted item management to retain specific entries during deletion.

These updates aim to enhance user experience and improve the overall functionality of the application.

### CODE REVIEW

File: Ginger/Ginger/Environments/AppsListPage.xaml.cs [hunk 1]
```diff
@@ -108,7 +108,8 @@ private void InitReleaseComboBox()
         private void SetGridView()
         {
             //Set the grid name
-            grdApps.Title = $"'{AppEnvironment.Name}' Environment Applications";
+            string grdEnvName = General.EscapeAccessKey(AppEnvironment.Name);
```
Comment (Maheshkale447) [inline Ginger/Ginger/Environments/AppsListPage.xaml.cs:111]: Create separate function for to sanitize and set grid title

File: Ginger/Ginger/Environments/GingerOpsEnvWizardLib/AddGingerOpsEnvPage.xaml.cs [hunk 2]
```diff
@@ -81,6 +81,7 @@ public void WizardEvent(WizardEventArgs WizardEventArgs)
             {
                 case EventType.Init:
                     mWizard = (AddGingerOpsEnvWizard)WizardEventArgs.Wizard;
+                    mWizard.mWizardWindow.FinishButton(false);
```
Comment (Maheshkale447) [inline Ginger/Ginger/Environments/GingerOpsEnvWizardLib/AddGingerOpsEnvPage.xaml.cs:84]: Change name to reflect functionality of the method

File: Ginger/GingerCoreCommon/External/Configurations/GingerOpsConfiguration.cs [hunk 3]
```diff
@@ -153,6 +156,14 @@ public override string ItemName
             }
         }
 
+        private void OnConfigChanged()
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/External/Configurations/GingerOpsConfiguration.cs:159]: change function name to InvalidateToken

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreCommon/External/Configurations/GingerOpsConfiguration.cs (hunk 3 above)
Comment (AmanPrasad43) [reply Ginger/GingerCoreCommon/External/Configurations/GingerOpsConfiguration.cs:159]: done

File: Ginger/Ginger/Environments/GingerOpsEnvWizardLib/AddGingerOpsEnvPage.xaml.cs (hunk 2 above)
Comment (AmanPrasad43) [reply Ginger/Ginger/Environments/GingerOpsEnvWizardLib/AddGingerOpsEnvPage.xaml.cs:84]: Changed to DisableFinishButton

File: Ginger/Ginger/Environments/AppsListPage.xaml.cs (hunk 1 above)
Comment (AmanPrasad43) [reply Ginger/Ginger/Environments/AppsListPage.xaml.cs:111]: Created a separate function

File: Ginger/GingerCoreNET/WizardLib/IWizardWindow.cs [hunk 4]
```diff
@@ -26,5 +26,6 @@ public interface IWizardWindow
         void Close();
         // void ShowDialog(int width = 800);
         void NextButton(bool isEnabled);
+        void DisableFinishButton(bool isEnabled);
```
[the remark this answers is not in the document]
Comment (AmanPrasad43) [reply Ginger/GingerCoreNET/WizardLib/IWizardWindow.cs:29]: done

File: Ginger/Ginger/UserControlsLib/ucGridView/ucGrid.xaml.cs [hunk 5]
```diff
@@ -793,10 +806,23 @@ private void btnDelete_Click(object sender, RoutedEventArgs e)
...
-                    RemoveFromLiteDB(o);
+                    if(!(o is EnvApplication application && application.GOpsFlag))
+                    {
+                        mObjList.Remove(o);
+                        RemoveFromLiteDB(o);
+                    }
+                    else
+                    {
+                       IsItemsDeleted = true;
+                    }
```
[the remark this answers is not in the document]
Comment (AmanPrasad43) [reply Ginger/Ginger/UserControlsLib/ucGridView/ucGrid.xaml.cs:820]: o should be an Envapplication

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**